packages/components/buttons/accessible-button/src/accessible-button.js

```diff
@@ -39,9 +45,18 @@ const AccessibleButton = React.forwardRef((props, ref) => {
 
   const isButton = !props.as || props.as === 'button';
 
-  const buttonProps = {
-    type: props.type,
-  };
+  let buttonProps = {};
+  if (isButton) {
+    buttonProps = {
+      type: props.type,
+    };
+  } else if (!props.isDisabled) {
```

Comment:
Nit: do-expression?

Reply:
It gets a little weird-looking with the object braces for the logic I am trying to achieve, am i doing this right?

```
const buttonProps = 
do {
   if (isButton) {
     {
      type: props.type,
    }
   }
  else if (!props.isDisabled) {
    {
      role: 'button',
      tabIndex: '0',
      onKeyPress: event => getIsEnterOrSpace(event) && handleClick(event),
    }
  }
   else {
  { }
 }
```

Reply:
True that also don't make it much simpler either.

Reply:
...just tried to avoid the let in a way.